src/list.c

```
@@ -0,0 +1,131 @@
+#include "list.h"
+
+void flist_push_front(s_flist_node **list, s_flist_node *node) {
+    if ((list != NULL) && (node != NULL)) {
+        node->next = *list;
+        *list = node;
+    }
+}
+
+void flist_pop_front(s_flist_node **list, f_list_node_del func) {
+    s_flist_node *tmp;
+
+    if (list != NULL) {
+        tmp = *list;
+        if (tmp != NULL) {
+            *list = tmp->next;
+            if (func != NULL) func(tmp);
```

to be checked at the begining of the function?
If Null, should probably return an error.....

---

Not necessarily, it would still pop it from the list, but it would not be freed which might be needed sometimes.

---

Oh, so we could have a memory leak here, because if `func` is NULL, we'll point on the 2nd element (`tmp->next`), and the old one (`tmp`) will not be freed, and then, be lost :thinking:

---

It could (if the node was dynamically allocated).
But it's not an error for the linked-list, and does not prevent it from working.
And it also allows us to pop a node from one list and add it to another without having to allocate a duplicate node and copy the old one.